app/assets/stylesheets/components/forms.scss

```diff
@@ -0,0 +1,188 @@
+@import '../vars/variables';
+
+%form-styling {
+  background-color: $smoke-00;
+  border: 1px solid $smoke-30;
+  appearance: none;
+  border-radius: $br-default;
+  transition: all $transition-props;
+
+  &-hover {
+    border-color: $smoke-40;
+  }
+
+  &-disabled {
+    color: $smoke-60;
+    background-color: $smoke-10;
+  }
+
+  &-checked {
+    background-color: $dolphin;
+    border-color: $dolphin-darker;
+    color: $snowflake;
+    background-position: center center;
+    background-repeat: no-repeat;
+  }
+
+  &-checked-disabled {
+    background-color: $dolphin-lighter;
+    border-color: $dolphin-lighter;
+    color: rgba($snowflake, 0.5);
+  }
+}
+
+// Text inputs and textareas.
+.crayons-textfield {
+  line-height: $lh-base;
+  padding: 0.5em;
+  font-family: inherit;
+  font-size: $fs-base;
+  width: 100%;
+  @extend %form-styling;
+
+  &::placeholder {
+    color: $smoke-60;
+  }
+
+  &:hover {
+    @extend %form-styling-hover;
+  }
+
+  &:focus {
+    background-color: $snowflake;
+    border-color: $dolphin;
+    box-shadow: 1px 1px 0 $dolphin;
+  }
+
+  &--disabled,
+  &[disabled] {
+    @extend %form-styling-disabled;
+  }
+
+  &::-ms-clear {
+    display: none;
+  }
+
+  &--icon {
+    padding-left: $su-7;
+  }
+}
+
+// Checkboxes, Radios
+.crayons-checkbox,
+.crayons-radio {
+  width: 1.125em;
+  height: 1.125em;
+  cursor: pointer;
+  background-position: center center;
+  vertical-align: middle;
+  @extend %form-styling;
+
+  &:hover {
+    @extend %form-styling-hover;
+  }
+
+  &--disabled,
+  &[disabled] {
+    @extend %form-styling-disabled;
+  }
+
+  &--checked,
+  &:checked {
+    @extend %form-styling-checked;
+    background-image: url("data:image/svg+xml,%3Csvg width='12' height='10' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath fill-rule='evenodd' clip-rule='evenodd' d='M11.157.933a.75.75 0 01.077 1.058L4.817 9.407a.75.75 0 01-1.134 0L.766 6.037a.75.75 0 011.135-.982L4.25 7.77l5.85-6.76a.75.75 0 011.057-.077z' fill='%23fff'/%3E%3C/svg%3E");
+
+    &--disabled,
+    &[disabled] {
+      @extend %form-styling-checked-disabled;
+    }
+  }
+}
+
+// Radios
+.crayons-radio {
+  border-radius: 50%;
+
+  &--checked,
+  &:checked {
+    @extend %form-styling-checked;
+    background-image: url("data:image/svg+xml,%3Csvg width='6' height='6' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Crect width='6' height='6' rx='3' fill='%23fff'/%3E%3C/svg%3E");
```

Comment:
Using background image like that, even though feels dirty, seems like the best approach to achieve what we want..

It makes most sense for checkboxes, because for radios I could do all the circle shapes with CSS only (no images), but I wanted to stay consistent.